Several fixes and performance improvement

ahmed2m opened the pull request:
1. Increased gap in feed buttons container.
2. Fix and encapsulate comment meta retrieval. (reddit changed the component)
3. Comments rendering was extremely slow with the dynamicElement helper, I made it async so it's way faster now.
4. Update flair styles.
5. Fix resources sidebar section being randomly expanded on navigating between subreddits/users.
6. Fix the show bookmarks functionality for posts.
7. Fix script crash in users' pages (when navigating to posts and comments tabs on users profiles)
8. Fix sidebar subs filter
9. Improve scroll to top button specially for smaller screens when wide-mode is off. (Added Floating Action Button for it for mobile screens)
10. Updated the release npm script to "prerelease" to sync both package and script versions, and it makes more sense that way, bump first then build new script.


<details><summary><h2>Screenshots and video:</h2></summary>
<p>


## Was:
https://github.com/user-attachments/assets/696d1388-d758-4ac1-9bef-988ff54a618a


## Now:


https://github.com/user-attachments/assets/af95beaa-a851-4aaf-bfcb-a444dc892295



</p>
</details>

ahmed2m committed:
increased gap in feed container and ensured consistent padding for feed buttons

ahmed2m committed:
fix and encapsulate comment meta retrieval

ahmed2m committed:
replace dynamicElement with observeOnce for improved comment rendering

ahmed2m committed:
update flair styles

ahmed2m committed:
improve sidebar resources section handling

ahmed2m committed:
refactor: fix context menu retrieval for posts

ahmed2m committed:
feat: enhance scroll-to-top functionality with FAB and adaptive sizing

TripleEYE commented:
As a follow-up to [our conversation on Greasyfork](https://greasyfork.org/en/scripts/490046-reddit/discussions/306747), I am pasting my related reply here as well, mostly to be able to attach [video screenshot](https://github.com/user-attachments/assets/50914601-ab35-4048-83ce-cae73a9c0da3).

I don't know if I'm confused here, or what.
I updated the script using the direct link (then re-installed, just in case, which reset to default settings I believe).

First, let me take a detour and request a feature to make the Profile Menu scrollable. Granted, I primarily use Reddit on a large monitor (at least when using on the PC), which I guess is a less common usage case, but I have to zoom out to 75% zoom or else I cannot see the Reddit++ Settings button. Making this menu scrollable would solve this and also solve access to other items towards the bottom of this menu.

Now, back to the topic at hand. Thank you for being on it! 🙏👍
Unfortunately, the 'Show Bookmark & Show Bookmark Always' on comments still doesn't work for me. The save button is still hidden within the kebab menu.
This feature has worked fine on the posts for me for a while now (I think even before I posted the above bug & feature report).

As for the collapsible Navigation menu on the left: It still has issues with memory after a page refresh, but it seems the Resources tab now DOES remember its collapsed state, and it is now the Custom Feeds, and the Communities tabs that forget they were collapsed.

I hopes the video helps somehow.

ahmed2m committed:
enhance: make user profile menu scrollable

ahmed2m commented:
Thanks! It is reddit seemingly making more updates to some elements that led to multiple breaks for the past two weeks. It just broke again after my fix. Anyhow, I updated the pull request on github, everything should be sorted out for now.

Would love your feedback, if anything is breaking for you

Pinging maintainer for review @lnm95

TripleEYE commented:
All sorted except for one little bug left that's encountered less due to its location. Details [on Greasyfork](https://greasyfork.org/en/scripts/490046-reddit/discussions/306747#comment-617131).

ahmed2m committed:
Update the release npm script to "prerelease" to sync both package and script versions

ahmed2m committed:
fix: subs filter

reddit moved the add the subreddit button outside the container and replaced it with manage subreddits

ahmed2m committed:
fix: frequent crashes due to nav bar variations

ahmed2m committed:
release v1.1.2

lnm95 commented:
Most of the issues are fixed in 1.2.
3. This requires some investigation — at minimum, in its current form it places the user rating in the wrong position. It might be better to simply rewrite `dynamicElement` and base it on a `MutationObserver`.
9. I fixed the bug related to `scrollToTop` (when you open comments and then return to the feed, the button breaks), so this requires resolving. It’s better to submit it as a separate branch/PR.
10. I believe I tried doing it this way originally and something didn't work in that pipeline. I’ll test it again.

ahmed2m commented on src/modules/header.less:
```diff
@@ -2,6 +2,11 @@
     text-decoration: none;
 }
 
+#user-drawer-content {
+    max-height: 90vh;
+    overflow: auto;
+}
+
```
@lnm95 Can you incorporate this change somehow? it makes the user drawer (popover when you click on the profile page) scrollable on zoomed in/small viewports. An issue reported by @TripleEYE

lnm95 replied:
Yeah, I saw that. Will you open a new PR, or should I do it myself?

ahmed2m replied:
Should be up at #11
